File: sdk/php/src/Internal/Infra/WsToken.php
<?php

namespace KuCoin\UniversalSDK\Internal\Infra;
use JMS\Serializer\Annotation\Exclude;
use JMS\Serializer\Annotation\SerializedName;
use JMS\Serializer\Annotation\Type;
class WsToken
{
    /**
     * @var string $token
     * @Type("string")
     * @SerializedName("$token")
     */
    public $token;
    /**
     * Websocket domain URL. It is recommended to use a dynamic URL, as the URL may change.
     * @var string $endpoint
     * @Type("string")
     * @SerializedName("endpoint")
     */
    public $endpoint;
    /**
     * Whether to encrypt. Currently only supports wss, not ws
     * @var bool $encrypt
     * @Type("bool")
     * @SerializedName("encrypt")
     */
    public $encrypt;
    /**
     * Network Protocol
     * - 'websocket' : Websocket
     * @var string $protocol
     * @Type("string")
     * @SerializedName("protocol")
     */
    public $protocol;
    /**
     * Recommended ping interval (milliseconds)
     * @var int $pingInterval
     * @Type("int")
     * @SerializedName("pingInterval")
     */
    public $pingInterval;
    /**
     * Heartbeat timeout (milliseconds)
     * @var int $pingTimeout
     * @Type("int")
     * @SerializedName("pingTimeout")
     */
    public $pingTimeout;

    public function __construct(
        $token = '',
        $pingInterval = 0,
        $endpoint = '',
        $protocol = '',
        $encrypt = false,
        $pingTimeout = 0
    )
    {
        $this->token = $token;
        $this->pingInterval = $pingInterval;
        $this->endpoint = $endpoint;
        $this->protocol = $protocol;
        $this->encrypt = $encrypt;
        $this->pingTimeout = $pingTimeout;
    }
}

File: sdk/java/src/main/java/com/kucoin/universal/sdk/internal/interfaces/WsToken.java
package com.kucoin.universal.sdk.internal.interfaces;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/** Web-socket token object returned by the “bullet” API. */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class WsToken {

  /** Token string used when establishing the socket. */
  @JsonProperty("$token")
  private String token;

  /** WebSocket domain URL (may change, always use the latest one). */
  @JsonProperty("endpoint")
  private String endpoint;

  /** Whether the endpoint is encrypted – currently only <code>wss://</code> is supported. */
  @JsonProperty("encrypt")
  private boolean encrypt;

  /** Network protocol, e.g. <code>websocket</code>. */
  @JsonProperty("protocol")
  private String protocol;

  /** Recommended ping interval in milliseconds. */
  @JsonProperty("pingInterval")
  private int pingInterval;

  /** Heart-beat timeout in milliseconds. */
  @JsonProperty("pingTimeout")
  private int pingTimeout;
}
